crash-recovery/README.md

```diff
@@ -0,0 +1,78 @@
+```
+shortname: ?/CRASH-RECOVERY
+name: Restore system state after crash
+type: standard
+status: raw
+editor: Vanshdeep Singh <vanshdeep@bigchaindb.com>
+```
+
+## Abstract
+Unexpect crashes are bound to occur when BigchainDB is deployed in production which could leave the system in an inconsistant state. Hence it is necessary that an appropriate methodology should be developed to restore the system to a consistant state.
+
+
+## Motivation
+For BigchainDB, a crash which can cause the state to be inconsistent could leave the system inoperable. The 'achilles heel' for BigchainDB is its `commit` function during which a new block with its transactions is written to the disk. Furthermore, UTXO merkle tree is updated based on these new transactions. An overview of what happens during `commit` is given below,
+
+1. Store metadata of the transactions.
+1. Store assets of the transactions.
+1. Store transactions and update UTXO.
+1. Store Block.
+
+These write operations are not transactional so any intermediate failure could result in an inconsistent state. For example, 
+
+- Consider a scenario where the system fails after step 2 i.e. we have a state where such metadata and assets exists which don't belong to any transaction (also called zombie assets and zombie metadata).
+- Consider a scenario wherein the system fails during setup 3 i.e. we would have a state which zombie metadata, zombie assets and zombie transactions (because the block to which these transactions belong is written in 4.). Moreover we might have an inconsistent UTXO.
+
+Hence it is crucial that the system should be able to recover from such crashes and bring itself to a consistent state.
```

Comment:
I wonder what Tendermint does about inconsistent state?

Reply:
@ttmc It is out of scope of this document to discuss how Tendermint handles inconsistent state but the issue pointed out by @vrde has some decent insights.

Reply:
ok